multi_partner_learning.py

```diff
@@ -451,12 +471,20 @@ def init_with_new_models(self):
 
         # Generate a new model and add it to the list
         new_model = self.generate_new_model()
-        partners_model_list.append(new_model)
 
-        # For each remaining partner, duplicate the new model and add it to the list
-        new_model_weights = new_model.get_weights()
-        for i in range(len(self.partners_list)-1):
-            partners_model_list.append(self.build_model_from_weights(new_model_weights))
+        # For each partner, create a new model and add it to the list
+        if isinstance(new_model, type(LogisticRegression())):
+            partners_model_list.append(new_model)
+            # For each remaining partner, create a new model and add it to the list
+            for i in range(len(self.partners_list)-1):
+                partners_model_list.append(self.generate_new_model())
+        else:
+            partners_model_list.append(new_model)
+
+            # For each remaining partner, duplicate the new model and add it to the list
+            new_model_weights = new_model.get_weights()
+            for i in range(len(self.partners_list)-1):
+                partners_model_list.append(self.build_model_from_weights(new_model_weights))  # is it really necessary to build it from weight instead of scratch ?
```

Comment:
Yes it is necessary: this allows having the exact same weights for each partner, and not have different models to start from (as `generate_new_model()` uses random initialization)

Reply:
Alright, comment removed ! Thanks